src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -25,6 +26,26 @@ type EncryptSubmissionBody = {
   version: number
 }
 
+type attachments = {
+  encryptedFile?: {
+    binary: string
+    nonce: string
+    submissionPublicKey: string
+  }
+}
+type EncryptSubmissionBodyAfterProcess = {
+  responses?: FieldResponse[]
+  encryptedContent: string
+  attachments?: attachments
+  isPreview: boolean
+  version: number
+  parsedResponses: ProcessedFieldResponse[]
+}
+
+type WithAttachmentsData<T> = T & { attachmentData: attachments }
+
+type WithFormData<T> = T & { formData: string }
```

Comment:
maybe extract these into a `types` file for reusability and to keep this file clean

Reply:
edited